src/web_app/static/js/reminder.js

```diff
@@ -0,0 +1,198 @@
+
+// classes disappear after applying filters and then sorting
+// so reset them after table body rendering
+$(document).ready(function (){
+	$('#reminders-table').on('post-body.bs.table', setClasses)
+})
+
+
+// passes all filter fields to the back
+function getFilters(){
+	let course = $('#course').val()
+	let author = $('#author').val()
+	let from_date = $('#from').val()
+	let to_date = $('#to').val()
+	let posted = $('#check').is(':checked')
+	$.post('/ui/reminder/filter?name=' + course + '&author=' + author + '&from=' + from_date + '&to=' + to_date + '&posted=' + posted,
+		updateTable, 'json')
+}
+
+/*
+updates data in the table
+params: data : json { result : number of reminders
+					  rows : reminder data }
+ */
+function updateTable(data){
+	let optionsHTML = "&hellip;<div class=\"dropdown-menu menu\">\n" +
+						"<a class=\"ml-3 mr-2\" href=\"#\" onclick=\"editReminder(this)\">Edit</a>\n" +
+						"<a class=\"mx-1\">or</a>\n" +
+						"<a class=\"mr-3 ml-2\" href=\"#\" onclick=\"deleteReminder(this)\">Delete</a>\n" +
+						"</div>"
+	for(let i=0; i<data.result; i++){
+		data.rows[i].posted = data.rows[i].posted ? 'True' : 'False'	// just for the capital letter
+		data.rows[i].options = optionsHTML	// add dropdown menu with edit and delete options
+	}
+	$('#reminders-table').bootstrapTable('load', data)
+	setClasses()	// method 'load' removes all class attributes
+}
+
+/*
+resets classes on all rows after data update
+make 5th column a dropdown button
+ */
+function setClasses(){
+	$("table#reminders-table tr").each(function() {
+		$("td", this).each(function(j) {
+			switch(j){
+				case 0:
+					this.setAttribute('class', 'reminder-id')
+					break;
+				case 1:
+					this.setAttribute('class', 'end_date')
+					break;
+				case 2:
+					this.setAttribute('class', 'timer')
+					break;
+				case 3:
+					this.setAttribute('class', 'text')
+					break;
+				case 4:
+					this.setAttribute('class', 'posted')
+					break;
+				case 5:
+					this.setAttribute('class', 'btn w-100')
+					this.setAttribute('data-toggle', 'dropdown')
+					break;
+			}
+		});
+	});
+}
+
+// get all reminders
+function resetFilters(){
+	$.post('/ui/reminder/filter', updateTable, 'json')
+}
+
+let rowDataBeforeEdit = { end_date:'', timer:'', text:'', posted:''}	// needed for saving row state in case of canceling
+function editReminder(eventObject){
+	// get index of row on which event was called
+	// path: a -> dropdown-menu -> td -> tr
+	let index = eventObject.parentElement.parentElement.parentElement.attributes[0].value
```

Comment:
jel se može dinamički postavit `id` remindera u tablici i onda se u `editReminder` stavlja taj `id` remindera i prek toga se radi brisanje, ovo nije robusno, ako se malo promijeni html logika ovdje pada

edit:
sad vidim da je riječ o nekakvom indexu po kojem se pristupa listama, to bi se moglo nekim for-om u htmlu riješit?

Reply:
vidim da ima još takvih linija, tak da i za njih vrijedi komentar

Reply:
problem je dohvat tog `id`a jer `this` vrati element unutar `dropdown menu`a koji onda nema pristup `id`u ćelije ili reda tako da se mora ići izvan njega.
kako misliš for-om? da prođe kroz sve redove i traži koji je pozvao i onda kad nađe spremi index?